enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,490 @@
+other namespaces. 
+
+Any owners/authors/users of Pods in those namespaces can opt into the consumption of the content of those shared `Secrets` / 
+ `ConfigMaps` during their Pod's execution.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.  Or if that PR is replaced by another proposal, or this proposal, this proposal will 
+point that out.
+
+
+## Motivation
+
+Simplify provision and consumption of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces. 
+
+### Goals
+
+- Provide easy to use sharing of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
+
+- When a shared `Secret` or `ConfigMap` changes, the container filesystem content the CSI driver injects into the Pod
+needs to reflect those changes.
+
+- When a cluster-admin revokes access to shared `Secrets` or `ConfigMaps` that as previously granted, that data should
+no longer be accessible.
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the `Secret` or `ConfigMap` is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal provides 
+for watching for updates to those objects.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update when new Pods consuming the content are provisioned.
+
+2. Protecting this content from being viewed/copied by users: if it's available for the Pod to use, the user who can 
+create a Pod that can utilize this feature can also read / view / exfiltrate the Secret/ConfigMap contents. It is technically 
+impossible to avoid that.
+
+3. This proposal for now assumes linux only for the underlying host/node.  No windows, etc. 
+
+## Proposal
+
+### User Stories [optional]
+
+1. As a cluster admin with an OpenShift subscription, I want to share the entitlement keys and configuration with my 
+users' workload Pods so that containers in those Pods can install RHEL subscription content.
+
+2. As a cluster admin with a set of CAs pertinent to various users's workloads, I want to share those CAs with my 
+users' workload Pods so that containers in those Pods can utilize those CAs when performing TLS based communication.
+
+3. As a cluster admin whose user workloads include kmod based scenarios, I want to share the necessary information to
+facilitate those scenarios across multiple namespaces.
+
+4. As a cluster admin who has to rotate credentials shared via this proposal, I want to be able to remove access to 
+stale versions of those credentials. 
```

Comment:
You want to remove access to new versions of the credentials(i.e. stop updating content within the consuming pod), not stale versions (the stale versions may have already been copied by the user in question anyway, so whether we delete that content out of the pod, or leave it in place, doesn't particularly matter from a security perspective.  Whether it's nicer to remove it or leave it, from the perspective of a user running the pod, open to debate i suppose).

Reply:
OK reconciling with https://github.com/openshift/enhancements/pull/250#discussion_r407620058

```
ask storage if removal is more normal than leaving stale content. I'm ok with either.

Describe the scenario we talked about with a cluster-admin trying to revoke access and rotate creds. That explains why it is important to make the SAR check.
```

Yeah, in hindsight I perhaps conflated those two points when I should not have.

Will update thanks.